### ISSUE TITLE
Question about what income is subject to the Self-Employment Tax

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/Tax-Calculator#1436]
I'm hoping to get some advice on which types of income are subject to the Self-Employment (payroll) Tax.

Since the early days of Tax-Calculator, the `functions.py` code subjects only two types of income to the Self-Employment Tax: (1) Schedule C self-employment net income (`e00900p` and `e00900s`) and (2) Schedule F farm net income (`e02100p` and `e02100s`).

But my reading of the IRS forms suggests there may be a third type of income that we are ignoring:  Schedule E: partnership and S-corporation net income/loss (`e26270`).

Here is the reason I suspect we are missing some income subject to the Self-Employment Tax. 
First, 2016 Schedule SE mentions **three** sources of income: Schedule C and Schedule F (as we have in the code) and Schedule K-1 (Form 1065), box 14.  Schedule K-1 is titled "Partner's Share of Income, Deductions, Credits, etc." and box 14 is labeled "Self-employment earnings (loss)".  So, clearly the amount in box 14 of Schedule K-1 is subject to the Self-Employment Tax.

The key question: What variable in the 2009 IRS-SOI PUF dataset corresponds to K-1 box 14?

It seems likely to me that `e26270`, Schedule E: partnership and S-corporation net income/loss, is what is put in Schedule K-1 box 14.  But I can't find any instructions that say this.

Can anybody shed light on this matter?  It's a pretty big issue in terms of tax revenue.

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @hdoupe @econ02 @codykallen

**Comment by jdebacker** [comment]
This is [indeed complicated](https://www.forbes.com/sites/anthonynitti/2016/10/03/irs-grows-increasingly-aggressive-in-subjecting-income-of-llc-member-to-self-employment-tax/#384367a678a8), but I think you are honing in on the best info available from the PUF.  However, my understanding is that passive income is not subject to SECA, so you may want to pull the non-passive income variables.  So net non-passive partnership income would be `e25980-e26110-e25960` and net non-passive S-corp income would be `e26190-e26100-e26180`.

**Comment by feenberg** [comment]
I think we might back out self-employment income from the FICA tax paid on 
self-employment income. I don't see any other way to tell if Schedule E is 
earnings or property income - it can be either.

dan

**Comment by martinholmer** [comment]
@jdebacker and @feenberg, Thank you both for your insightful comments on issue #1436.  I am going to look into the 2009 PUF variables you mention, but only after I'm back from Canada on Tuesday, June 27th.

**Comment by martinholmer** [comment]
With respect to issue #1436, I've looked in the taxdata repo at the current version of the private `cps-matched-puf.csv` file used as input to the `pufdata/finalprep.py` script.  That file contains all the variables needed to explore the suggestions made by @jdebacker and @feenberg.  I'll begin working on this issue when I return from vacation on Tuesday, June 27th. 

It would seem that this issue has identified a bug in Tax-Calculator logic, in that some income subject to the self-employment tax is not being included in the calculations in the `EI_PayrollTax` function in the `functions.py` file.  Fixing this bug will require new variables in the `puf.csv` file as well as new `functions.py` logic that uses those variables to compute a more complete estimate of the self-employment tax.

@MattHJensen @feenberg @jdebacker @Amy-Xu @andersonfrailey @hdoupe

**Comment by martinholmer** [comment]
@jdebacker said in the issue #1436 conversation:
> This is indeed complicated, but I think you are honing in on the best info available from the PUF. However, my understanding is that passive income is not subject to SECA, so you may want to pull the non-passive income variables. So net non-passive partnership income would be e25980-e26110-e25960 and net non-passive S-corp income would be e26190-e26100-e26180.

Thanks, @jdebacker, for the plausible suggestion.  However, I can't see any relationship between the six variables you mention and the amount in Schedule K-1 box 14, which is titled `Self-employment earnings (loss)`.  I've looked a quite a few filing units and don't see any connection.

Below are three very simple filing units where there is no wage/salary income (e00200), no Sch C self-employment income (e00900), and no Sch F self-employment income (e02100), even though the 2009 PUF says they have positive self-employment income subject to the SECA tax calculated on Sch SE.  These three filing units are even simpler than what I've said so far: they have zero taxable income under the SECA for the taxpayer and positive SECA taxable income only for the spouse.

The reason to look at these three filing units is that we do know from a PUF variable (e30500) how much spouse self-employment income is subject to the SECA (self-employment) tax.  And because all the other sources of self-employment income are zero in these cases, it seems as if the e30500 value (which is denoted by k30450s in the code below) must be equal to the amount in Sch K-1 box 14.

Here is the Python code that defines these variables and then attributes of the three filing units.
First the Python code:
```
def split_earnings_variables(data):
    """
    Split earnings subject to SECA tax between taxpayer and spouse
    """
    total_wages = np.where(data['MARS'] == 2,
                           data['wage_head'] + data['wage_spouse'], 0)
    wage_frac_p = np.where(total_wages != 0,
                           data['wage_head'] / total_wages, 1.)
    wage_frac_s = 1.0 - wage_frac_p
    data['e00200p'] = wage_frac_p * data['e00200']
    data['e00200s'] = wage_frac_s * data['e00200']
    # old e00900 and e02100 splitting logic:
    data['e00900p'] = wage_frac_p * data['e00900']
    data['e00900s'] = wage_frac_s * data['e00900']
    data['e02100p'] = wage_frac_p * data['e02100']
    data['e02100s'] = wage_frac_s * data['e02100']
    # SECA taxable income capped at 2009 MTE of 106800 dollars
    data['secatip'] = data['e30400'] - data['e30500']
    data['secatis'] = data['e30500']
    # JD-suggested variables
    data['ktot1'] = data['e25980']
    data['ktot2'] = data['e26190']
    data['k1791'] = data['e26110']
    data['k1792'] = data['e26100']
    data['klss1'] = data['e25960']
    data['klss2'] = data['e26180']
    return data
```
And now the attributes of these three filing units:
```
sqlite> select 
          secatis, ktot1, ktot2, k1791, k1792, klss1, klss2
        from data
        where e00200=0 and e00900=0 and e02100=0 and secatip=0 and
              secatis>50000 and secatis<90000; 
78100|0|0|0|0|0|0
61200|117500|777600|0|0|2360000|0
53100|199500|35000|0|0|610|10500
```
Maybe I'm missing something here, but it doesn't look like these six variables are very helpful in estimating the values in Sch K-1 box 14 for the taxpayer and spouse.

**Comment by jdebacker** [comment]
@martinholmer Thanks for digging into the data to confirm this.  I can't explain the discrepancies, but some things to consider:
1) S-Corp business income does not trigger any SECA liability. 
2) Passive income from partnerships is excluded (at least in the case of it flowing to a limited partner, not 100% sure what the rule is for general partners - see link above and [here](https://www.irs.gov/individuals/tax-trails-self-employment-income-8)).
3) Limited partners' income from guaranteed payments are subject to SECA, but their distribution of business profits are not.
4) The rules aren't entirely clear to me, so it's hard to know how much error filers are making.  It's not an error in this sense, but there's at least [anecdotal evidence that some filers who should know better aren't paying they SECA tax they ought to](https://www.forbes.com/sites/anthonynitti/2016/08/12/clintons-paid-3-6-million-in-federal-tax-in-2015-could-should-have-paid-a-lot-less/#1d0c381269e0).

Re (4), the differences between net non-passive partnership income and what is reported on Line 1a of Schedule SE (`e30400`) seem too large to be error.  Perhaps a higher fraction of passive income flows to general partners than I thought and this enters into `e30400`?

Does it work for your purposes to just take Line 1a of Schedule SE then subtract off Schedule F income?

**Comment by martinholmer** [comment]
@jdebacker said:
> Re (4), the differences between net non-passive partnership income and what is reported on Line 1a of Schedule SE (e30400) seem too large to be error. Perhaps a higher fraction of passive income flows to general partners than I thought and this enters into e30400?

The 2009 SOI PUF variables `e30400` and `e30500` are **not** line 1a of Schedule SE.  Here is how they are defined in the SOI documentation:
```
E30400 --- NET EARNINGS FROM SELF-EMPLOYMENT INCOME
The lesser of Schedule SE Section B, Part I line 6 (net Schedule SE earnings) or
Part I line 9 (maximum amount subject to Social Security tax) on the combined
Schedule SE. For Section A of Schedule SE, this is line 4. The combined
Schedule SE is the sum of the data found on the Schedule SE of the primary
taxpayer and the Schedule SE of the secondary taxpayer.

E30500 --- NET EARNINGS FROM SELF-EMPLOYMENT INCOME, SECONDARY TAXPAYER
The lesser of Schedule SE, Section B, Part I line 6 (net Schedule SE earnings) or
Part I line 9 (maximum amount subject to Social Security tax) on the secondary
taxpayer’s Schedule SE. For Section A of Schedule SE, this is line 4.
```

@jdebacker continued:
> Does it work for your purposes to just take Line 1a of Schedule SE then subtract off Schedule F income?

Well that would be easy, but Line 1a doesn't seem to be a variable in the 2009 SOI PUF dataset.

I think the only approach left is to estimate (a lower bound) for Schedule K-1 box 14 (which we know is included in Schedule SE taxable income) by using a variant of the suggestion @feenberg made earlier in the #1436 discussion.  He suggested backing out an estimate of the amount in Schedule K-1 box 14 from the the SE tax liability (e09400) for the whole filing unit.  But we can avoid some problems inherent in that approach because we have `e30400` and `e30500`.  This means we have (the capped) self-employment income subject to Schedule SE taxation for both the taxpayer and for the spouse (if present).  This approach implies we will never estimate a 2009 value for Schedule K-1 box 14 that is larger than $106,800.  But that is still better than assuming that Schedule K-1 box 14 is always zero, which is what we are doing now.

Further comments and suggestions are welcome.

**Comment by martinholmer** [comment]
Issue #1436 has been resolved by the merge of pull request #1438.
